include/fastdds/rtps/RTPSDomain.h

```diff
@@ -103,6 +104,22 @@ class RTPSDomain
             WriterHistory* hist,
             WriterListener* listen = nullptr);
 
+    /**
+     * Create a RTPSWriter in a participant.
+     * @param p Pointer to the RTPSParticipant.
+     * @param watt Writer Attributes.
+     * @param payload_pool Shared pointer to the IPayloadPool
+     * @param hist Pointer to the WriterHistory.
+     * @param listen Pointer to the WriterListener.
+     * @return Pointer to the created RTPSWriter.
+     */
+    RTPS_DllAPI static RTPSWriter* createRTPSWriter(
+            RTPSParticipant* p,
+            WriterAttributes& watt,
```

Comment:
```suggestion
            const WriterAttributes& watt,
```

Reply:
`RTPSParticipantImpl` is not const-correct, and it expects a non-const `WriterAttributes`. The same goes for the methods that `RTPSParticipantImpl`calls, increasing the ramifications of the change.

We can keep the signature of the method as the old one (with non-const `WriterAttributes`) or `const_cast` it before calling `RTPSParticipantImpl`

Reply:
Keeping the old signature is ok